### PR TITLE
Update read_habitatmap, read_hm_stdized ,read_hm_terr for hm 2020

### PR DESCRIPTION
**Pull request by cecileherr**
Updated: 

- read_habitatmap()
- read_habitatmap_stdized() 
- read_habitamap_terr()

For **read_habitatmap()** : added a basic control of the checksums to ensure that the used version for habitatmap corresponds to the version of habitatmap_stdized stored under n2khab_data (useful since read_habitatmap_stdized() is being called within read_habitatmap())

About the **errors in geometry** in habitatmap and the processed versions (Ring Self-intersections, see also https://github.com/inbo/n2khab-preprocessing/pull/50#issuecomment-815528848) : 
 Despite the errors, geometric operations are possible with these layers: e.g. an intersection seems to produce the expected results ([more info](https://github.com/inbo/n2khab-preprocessing/pull/50#issuecomment-838423661)). Using `st_make_valid` has not been implemented in the read_ functions (my guess is that it is not needed with the present version of sf) but it could be an option for read_hm_terr and read_hm_stdized since:

- it seems to remove the errors - no errors anymore with st_valid -, 
- the faulty polygons before and after seem ok (though not exhaustively checked)
- st_make_valid costs only a few seconds (<5). 

Using st_make_valid for the raw version of the habitatmap takes longer though (1 min). Might be better not to use it since read_habitatmap() takes already quite long to run.

### CODE REVIEW

**Comment by florisvdh** [comment]
@cecileherr thank you, will review the function updates asap!

Responding already to `st_make_valid()`:

> it could be an option for read_hm_terr and read_hm_stdized since

If you would like to implement `st_make_valid()` for `habitatmap_terr` and `habitatmap_stdized`, then I suggest to update the data sources themselves in the PR where they're made (see https://github.com/inbo/n2khab-preprocessing/pull/50#issuecomment-838494498), not doing it in the reading functions.
- They were not merged yet (nor uploaded to Zenodo) because of potential updates with relation to `n2khab` (this PR).
- When doing it in the data sources, making polygons valid is unneeded at reading time. (I regard seconds as costly since user scripts are often rerun a lot, interactively.)
- You did similar updates in https://github.com/inbo/n2khab-preprocessing/pull/54.

Maybe you mean to provide it through an optional argument, but I think this would not help standardization either.

File: R/read_habitatdata.R [hunk 1]
```diff
@@ -797,8 +805,36 @@ read_habitatmap <-
...
+            }
+
+            if (!(xxh64sum_habitatmap_stdized_present %in%
+                xxh64sum_habitatmap_stdized_expected)) {
+                stop("You are trying to use habitatmap version '", version,"' ",
+                     "with another version of habitatmap_stdized. ",
+                     "Specify the correct version as argument (version =) ",
+                     "and add the corresponding files under ",
+                     "'n2khab_data/10_raw/habitatmap' and ",
+                     "'n2khab_data/10_processed/habitatmap_stdized'.",
```
Comment (florisvdh) [inline R/read_habitatdata.R:826]: Should use `20_processed`. Covered already in PR #140, where these lines were moved.

**Review by florisvdh — COMMENTED** [review]
Nice work @cecileherr ! :+1:

I suggest a few changes/updates in PR #140. There, the functions have been checked in a reprex as well. They work well!

Maybe you didn't check the bullets of #117? Apart from the last bullet (the consideration of which I leave to you - seems more of a detail to me), I tackled them in the referred PR (partly different from suggestions in the issue). Please have a look at its discussion and let me know what you think.

File: R/read_habitatdata.R [hunk 2]
```diff
@@ -120,9 +121,10 @@
 read_habitatmap_stdized <-
     function(file = file.path(fileman_up("n2khab_data"),
                               "20_processed/habitatmap_stdized/habitatmap_stdized.gpkg"),
-             version = "habitatmap_stdized_2018_v2"){
+             version = c("habitatmap_stdized_2020_v1",
+                         "habitatmap_stdized_2018_v2", "habitatmap_stdized_2018_v1")){
```
[the remark this answers is not in the document]
Comment (cecileherr) [reply R/read_habitatdata.R:125]: done in PR#[142](https://github.com/inbo/n2khab/pull/142)

**Comment by cecileherr** [comment]
> @cecileherr thank you, will review the function updates asap!
> 
> Responding already to `st_make_valid()`:
> 
> > it could be an option for read_hm_terr and read_hm_stdized (...)
>

About using `st_make_valid()` while creating the data sources: we decide [NOT to do it](https://github.com/inbo/n2khab-preprocessing/pull/50#issuecomment-839744510) for the moment.

**Review by florisvdh — APPROVED** [review]
Thanks a lot, well done. :tada: 

Will now make a separate PR with some more changes on the term 'vegetation', but it will probably target `dev_0.5.0`, you don't need to await it.

**Comment by florisvdh** [comment]
@cecileherr were you ready with this PR? If so, I'll just await the checks running here and then I'll merge.

**Comment by cecileherr** [comment]
> @cecileherr were you ready with this PR? If so, I'll just await the checks running here and then I'll merge.

Yes, this PR was ready (all the needed changes have already been made in PR#140)

**Pull request merged**